src/core/qgsfileutils.h

```diff
@@ -87,6 +87,19 @@ class CORE_EXPORT QgsFileUtils
      * \since QGIS 3.2
      */
     static QString findClosestExistingPath( const QString &path );
+
+    /**
+     * Will check \a basepath in an outward spiral up to \a maxDepth levels to check if \a fileName exists.
+     * \param fileName Name of the file to find
+     * \param basepath current basepath of the file
+     * \param maxDepth limit the number of time the search can move up
+     * \param driveMargin limits how close the search can get to the drive level
+     * \param currentDir alternative default directory to override the actual current directory during the search
+     * \returns List of strings of the first matching path in unix format.
+     * \since QGIS 3.12
+     */
+    static QList<QString> findFile( const QString &fileName, const QString &basepath, int maxDepth = 4, int driveMargin = 3, const QString &currentDir = "" );
```

Comment:
```suggestion
    static QStringList findFile( const QString &fileName, const QString &basepath, int maxDepth = 4, int driveMargin = 3, const QString &currentDir = "" );
```

this is a drop in replacement for `QList<QString>` (inherits from it) and has some nice additions.

Reply:
I always though it was an alias.